core/src/main/java/io/questdb/cairo/wal/CopySegmentFileJob.java

```diff
@@ -0,0 +1,147 @@
+            long rowCount,
+            LongList newColumnFiles,
+            int columnIndex
+    ) {
+        Path newSegPath = Path.PATH.get().of(walPath).slash().put(newSegment);
+        int setPathRoot = newSegPath.length();
+        dFile(newSegPath, columnName, COLUMN_NAME_TXN_NONE);
+        long primaryFd = openRW(ff, newSegPath, LOG, options);
+        newColumnFiles.setQuick(columnIndex * NEW_COL_RECORD_SIZE, primaryFd);
+
+        long secondaryFd;
+        if (ColumnType.isVariableLength(columnType)) {
+            iFile(newSegPath.trimTo(setPathRoot), columnName, COLUMN_NAME_TXN_NONE);
+            secondaryFd = openRW(ff, newSegPath, LOG, options);
+            newColumnFiles.setQuick(columnIndex * NEW_COL_RECORD_SIZE + 3, secondaryFd);
+        } else {
+            secondaryFd = -1;
+        }
+
+        boolean success;
+        if (ColumnType.isVariableLength(columnType)) {
+            success = copyVarLenFile(ff, primaryColumn, secondaryColumn, primaryFd, secondaryFd,  rowOffset, rowCount, newColumnFiles, columnIndex);
+        } else if (columnType > 0) {
+            success = copyFixLenFile(ff, primaryColumn, primaryFd, rowOffset, rowCount, columnType, newColumnFiles, columnIndex);
+        } else {
+            success = copyTimestampFile(ff, primaryColumn, primaryFd, rowOffset, rowCount, newColumnFiles, columnIndex);
+        }
+
+        if (!success) {
+            throw CairoException.critical(ff.errno()).put("failed to copy column file to new segment" +
+                            " [path=").put(newSegPath)
+                    .put(", column=").put(columnName)
+                    .put(", rowOffset=").put(rowOffset)
+                    .put(", rowCount=").put(rowCount)
+                    .put(", columnType=").put(columnType).put("]");
+        }
+    }
+
+    private static boolean copyVarLenFile(FilesFacade ff, MemoryMA primaryColumn, MemoryMA secondaryColumn, long primaryFd, long secondaryFd, long rowOffset, long rowCount, LongList newOffsets, int columnIndex) {
+        long indexMapSize = (rowOffset + rowCount + 1) * Long.BYTES;
+        long srcIndexAddr = TableUtils.mapRW(ff, secondaryColumn.getFd(), indexMapSize, MEMORY_TAG);
+        try {
+            long varStart = Unsafe.getUnsafe().getLong(srcIndexAddr + rowOffset * Long.BYTES);
+            long varEnd = Unsafe.getUnsafe().getLong(srcIndexAddr + (rowOffset + rowCount) * Long.BYTES);
+            long varCopyLen = varEnd - varStart;
+            boolean success = ff.copyData(primaryColumn.getFd(), primaryFd, varStart, varCopyLen) == varCopyLen;
+            if (!success) {
+                return false;
+            }
+
+            newOffsets.setQuick(columnIndex * NEW_COL_RECORD_SIZE + 1, varStart);
+            newOffsets.setQuick(columnIndex * NEW_COL_RECORD_SIZE + 2, varCopyLen);
+
+            long indexLen = (rowCount + 1) * Long.BYTES;
+            long dstIndexAddr = TableUtils.mapRW(ff, secondaryFd, indexLen, MEMORY_TAG);
```

Comment:
I'm adding `POSIX_MADV_SEQUENTIAL`, `src` and `dst` memory maps are read / written sequentially.

Reply:
Is `dst` ever read?

Reply:
no, it's written to

Reply:
Then we should use `POSIX_MADV_RANDOM` for it to avoid phantom disk reads. `POSIX_MADV_SEQUENTIAL` for `src` is not required as the default read-ahead is almost the same as MADV_SEQUENTIAL.

Reply:
ok